### DISCUSSION TITLE
Introduce release note automation

### DISCUSSION THREAD

**Discussion by trombik** [UncleRus/esp-idf-lib#334] [Ideas]
i was looking for a solution to automate releasing. there are several to choose from, and here is some I have looked at closely.

* [release-please](https://github.com/googleapis/release-please)
* [release-drafter](https://github.com/release-drafter/release-drafter)

`release-please` is a GItHub Action to automate releasing. Creates release notes, increment semver according to `git log`. I like the idea, but the implementation is complicated, and the documentation is, like other Google products, not very user-friendly. It took me hours to understand it. Testing and debugging are not very easy because, to test, you actually need to release. It is a complete automated solution if, and only if, you understand and follow their conventions.

`release-drafter` is a GitHub Action to automate writing release notes. It does not release, but create a _draft_ version of release notes. every time a commit is merged to the default branch, the action runs, and create a draft. you can review, or manually edit, the draft. as it does not do releasing, you need to release it. to automate further, some conventions need to be introduced. branch names for PRs must be prefixed with `fix-`, `feat-` or `doc-`, such as `fix-issue-1`. if you follow this convention, when the PR is created, the PR is labeled by another GitHub Action. when it is merged, the PR title is automatically included in the draft release note, categorized by the branch prefix.

when the repository is split into multiple repositories, releasing would be a pain. I would like to automate the process, and `release-drafter` looks not so invasive for us.

what do you think?

p.s. you can see how i implemented `release-drafter` at https://github.com/trombik/esp_hass

**Comment by UncleRus** [comment]
I like the second option (release-drafter). First of all, because it will discourage me from the bad habit of pushing minor edits directly to the master :)
